### PR TITLE
[libc] [doc] explain the need of `clang`

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
This patch mentions some more details of the infra of libc.

### CODE REVIEW

File: libc/docs/full_host_build.rst [hunk 1]
```diff
@@ -60,9 +60,12 @@ to use the freshly built lld and compiler-rt.
 We will go over some of the special options passed to the ``cmake`` command
 above.
 
-* **Enabled Projects** - Since we want to build and install clang, lld
-  and compiler-rt along with the libc, we specify
-  ``clang;libc;lld;compiler-rt`` as the list of enabled projects.
+* **Enabled Projects** - Since we want to build and install ``clang``, ``lld``
+  and ``compiler-rt`` along with the ``libc``, we specify
+  ``clang;libc;lld;compiler-rt`` as the list of enabled projects. ``clang`` is
```
Comment (SchrodingerZhu) [inline libc/docs/full_host_build.rst:65]: Separate runtime builds into `-DLLVM_ENABLE_RUNTIMES="libc;compiler-rt"`?

Comment (SchrodingerZhu) [reply libc/docs/full_host_build.rst:65]: > Is compiler-rt a runtimes project?

I think so.

> the only dependency I was aware of was the hdr-gen tool

I tried build without `clang`, but somehow it complains about missing symbols from my system LLVM.

Comment (jhuber6) [reply libc/docs/full_host_build.rst:65]: I've always enabled it as a project.

I do remember hearing some difficulties with `compiler-rt`, but I just attributed that to needing a full compiler toolchain to build it, so if you have a system installation of `clang` or `gcc` it should work without specifically needing the `compiler-rt` from in-tree.

Comment (SchrodingerZhu) [reply libc/docs/full_host_build.rst:65]: Isn't the case that we need to customize `scudo` to use it as our `libmalloc`?

Comment (michaelrj-google) [reply libc/docs/full_host_build.rst:65]: we can build with GCC, but we also need headergen to generate the headers and that currently is build with tablegen. It is possible to just build it separately and then use it with `-DLIBC_HDRGEN_EXE=/path/to/hdrgen` in your cmake.

As for TableGen, I'm pretty sure it's in `llvm/utils/TableGen`, so including `clang` shouldn't effect it either way.

Comment (michaelrj-google) [reply libc/docs/full_host_build.rst:65]: yes, I think the runtimes build can't build headergen right now. I'm not sure if we should update the commands to recommend using the runtimes build yet, due to the specific issue you ran into of not being able to build headergen. Once we have a standalone headergen then it seems like a good idea.

**Comment by SchrodingerZhu** [comment]
Closing this for now.

**Pull request closed without merging**